lib/active_record/virtual_attributes/virtual_total.rb

```diff
@@ -77,6 +88,14 @@ def define_virtual_aggregate_attribute(name, relation, method_name, column, opti
         end
       end
 
+      %w[average maximum minimum].each do |m_name|
+        define_method("define_virtual_#{m_name}_method") do |name, relation, column|
```

Comment:
fun: this define method and metadata makes me squeamish.

Fun because everything here is using metadata to define_method.